prepare/formats/empty_input_output_separators.py

```diff
@@ -1,6 +1,9 @@
 from src.unitxt.catalog import add_to_catalog
-from src.unitxt.formats import ICLFormat
+from src.unitxt.operators import ModelInputFormatter
 
-format = ICLFormat(input_output_separator="")
+format = ModelInputFormatter(
+    demo_format="{source}{target}\n\n",
+    model_input_format="{instruction}\n{demos}\n{source}",
```

Comment:
This adds 3 \n after the last demo, and two \n after the instruction if we don't have demos.

Maybe it should be :

demo_format="{source}{target}\n\n",
model_input_format="{instruction}\n{demos}{source}",

We need to image the output without demos.

Reply:
Now we also need to consider what happens if instruction is empty.   In this case we don't want the \n after it.

We need some to note that \n is only if the instruction is not empty.

{instruction+\n}{demos}{source}

such that if we find {XXX+something} we replace XXX with the value and append the something.

Do you have other ideas?

Reply:
no three \n any longer, because I removed the automatic appending of a non empty instruction with \n\n.  If we are talking about a fix prefix, like "Instruction: ", then I can certainly prepend with it, and append any non-"" instruction with one or two \n.

Reply:
I think this should be like the demo format. so you have an `instruction_format` argument which can help modifying it just like the `demo_format`.

Reply:
I thought that the aim is to reduce the number of formatting arguments. And since instruction is a simple string, why not, indeed, let the user wrap it with any formatting (prefix and or suffix) as they want?
I think that more important (or that I less understand) is the question of how to push that instruction (in my view: a whole formatted instruction) into the final result: by using `AddFields` a couple of Operation up the recipe (as is done now, but makes @matanor feel uncomfortable) or have an api to update `ModelInputFormatter`. I understood from @yoavkatz that an input argument is not possible once the Operator is a formal, catalogged, artifact.